src/main/java/com/evanlennick/retry4j/config/RetryConfig.java

```diff
@@ -3,67 +3,62 @@
 import com.evanlennick.retry4j.backoff.BackoffStrategy;
 
 import java.time.Duration;
-import java.util.HashSet;
 import java.util.Set;
 import java.util.function.Function;
 
 public class RetryConfig {
 
-    private Boolean retryOnAnyException = false;
-    private Set<Class<? extends Exception>> retryOnSpecificExceptions = new HashSet<>();
-    private Set<Class<? extends Exception>> retryOnAnyExceptionExcluding = new HashSet<>();
-    private Integer maxNumberOfTries;
-    private Duration delayBetweenRetries;
-    private BackoffStrategy backoffStrategy;
-    private Object valueToRetryOn;
-    private Boolean retryOnValue = false;
-    private Function<Exception, Boolean> customRetryOnLogic;
-    private boolean retryOnCausedBy;
+    private final boolean retryOnAnyException;
+    private final Set<Class<? extends Exception>> retryOnSpecificExceptions;
+    private final Set<Class<? extends Exception>> retryOnAnyExceptionExcluding;
+    private final Integer maxNumberOfTries;
+    private final Duration delayBetweenRetries;
+    private final BackoffStrategy backoffStrategy;
+    private final Object valueToRetryOn;
+    private final boolean retryOnValue;
+    private final Function<Exception, Boolean> customRetryOnLogic;
+    private final boolean retryOnCausedBy;
+
+    RetryConfig(boolean retryOnAnyException, Set<Class<? extends Exception>> retryOnSpecificExceptions,
+                Set<Class<? extends Exception>> retryOnAnyExceptionExcluding, Integer maxNumberOfTries,
+                Duration delayBetweenRetries, BackoffStrategy backoffStrategy, Object valueToRetryOn,
+                boolean retryOnValue, Function<Exception, Boolean> customRetryOnLogic, boolean retryOnCausedBy) {
+        this.retryOnAnyException = retryOnAnyException;
```

Comment:
i was actually thinking about starting to include Lombok into this project to help generate obnoxious getter/setter and constructor stuff like this... im fine if you want to keep in manual for now but just something to consider if you've never used it and might be interested: https://projectlombok.org/features/all

Reply:
sure, we could address in a new issue